src/main/java/com/avengers/gamera/service/ArticleService.java

```diff
@@ -1,32 +1,98 @@
 import org.springframework.stereotype.Service;
 
+import java.awt.print.Pageable;
+import java.util.List;
+
 @Service
 @RequiredArgsConstructor
 @Slf4j
 public class ArticleService {
     private final ArticleRepository articleRepository;
+    private final GameRepository gameRepository;
+    private final UserRepository userRepository;
     private final ArticleMapper articleMapper;
 
     public ArticleGetDto createArticle(ArticlePostDto articlePostDto) {
 
         Article article = articleMapper.articlePostDtoToArticle(articlePostDto);
         String img = article.getCoverImgUrl();
-
-        //TODO after image upload function implemented, this url should from user
         if (StringUtils.isBlank(img)) {
             article.setCoverImgUrl("https://spicsum.photos/800/400");
         }
-        log.info("Saving new article to database");
+        Game game = gameRepository.findById(articlePostDto.getGameId()).orElseThrow(()->
+                new ResourceNotFoundException("Related game")
+        );
+        article.setGame(game);
+        User user = userRepository.findById(articlePostDto.getAuthorId()).orElseThrow(() ->
+                new ResourceNotFoundException("Related Author(user)")
+        );
+        article.setUser(user);
+        log.info("==========Saving new article to database==============");
+        return articleMapper.articleToArticleGetDto(articleRepository.save(article));
+    }
+
+    public List<MiniArticleGetDto> getMiniArticles(int pageNumber, int pageSize){
+        log.info("Try to get some articles that has not been deleted .....");
+        List<Article> articles = articleRepository.findArticleByIsDeletedFalse((Pageable) PageRequest.of(pageNumber,pageSize));
+
+        if(articles.size()==0) {
+            return null;
+        }
+        return articleMapper.articleToMiniArticleGetDto(articles);
+
+    }
+
+    public List<MiniArticleGetDto> getMiniArticlesByType(ArticleType articleType, int pageNumber, int pageSize) {
+        log.info("Try to get some articles that has not been deleted .....");
+        List<Article> articles = articleRepository.findArticlesByTypeAndIsDeletedFalse(articleType, (Pageable) PageRequest.of(pageNumber, pageSize));
+        if(articles.size()==0) {
+            return null;
+        }
+        return articleMapper.articleToMiniArticleGetDto(articles);
+    }
+
+    public ArticleGetDto getArticleById(Long articleId) {
+        Article article = articleRepository.findById(articleId).orElseThrow(() ->
+                new ResourceNotFoundException("Related Article with the ID(" + articleId + ")")
+        );
+        return articleMapper.articleToArticleGetDto(article);
+    }
+
+    public void deleteArticleById(Long articleId) {
+        Article article = articleRepository.findById(articleId).orElseThrow(() ->
+                new ResourceNotFoundException("Related Article with the ID(" + articleId + ")")
+        );
+        articleRepository.delete(article);
```

Comment:
Do not literally delete a record from db. We have an attribute named `deleted` or `isDeleted` to implement soft deleting. Besides, even if you do delete it from db, you should not talk to Database twice.